File: PRACTICA1/RUBY/Deepspace/test_p1.rb
require './Dice.rb'
require './Loot.rb'
require './GameCharacter'
require './ShieldBooster'
require './SuppliesPackage'
require './Weapon'
require './WeaponType'
puts "Vamos a crear un objeto de cada clase y lo mostraremos"

dado1 = Dice.new()
puts "Mostramos los atributos del objeto de la clase Dice"
puts dado1.NHANGARSPROB, dado1.NSHIELDSPROB, dado1.NWEAPONSPROB, dado1.FIRSTSHOTPROB, dado1.generator
puts "Mostramos los metodos del objeto de la clase Dice"
puts dado1.initWithNHangars
puts dado1.initWithNWeapons
puts dado1.initWithNShields
puts dado1.whoStarts(4)
puts dado1.firstShot
puts dado1.spaceStationMoves(0.5)

botin = Loot.new(1,2,3,4,5)
puts "Mostramos los atributos del objeto de la clase Loot"
puts botin.nSupplies, botin.nWeapons, botin.nShields, botin.nHangars, botin.nMedals

escudo = ShieldBooster.new("nombre", 3.5, 3)
puts "Mostramos los atributos del objeto de la clase ShieldBooster"
puts escudo.name, escudo.boost, escudo.uses
puts "Mostramos métodos"
escudo_copia = ShieldBooster.newCopy(escudo)
puts "El objeto creado es: "
puts escudo_copia.uses, escudo_copia.name, escudo_copia.boost
puts "Llamamos a useIt"
puts escudo.useIt
puts escudo.uses

sumin = SuppliesPackage.new(2.3, 4.5, 1.2)
puts "Mostramos los atributos del objeto de la clase SuppliesPackage"
puts sumin.ammoPower, sumin.fuelUnits, sumin.shieldPower
sumin_copia = SuppliesPackage.newCopy(sumin)
puts "Mostramos el objeto copiado: "
puts sumin_copia.ammoPower, sumin_copia.fuelUnits, sumin_copia.shieldPower

arma = Weapon.new("pistola", WeaponType::LASER, 4)
puts "Mostramos los atributos del objeto de la clase Weapon"
puts arma.name, arma.type, arma.uses
arma_copia = Weapon.newCopy(arma)
puts arma_copia.name, arma_copia.type, arma_copia.uses

puts "Mostramos el poder del arma"
puts arma.power

puts "Probamos metodo useIt"
puts arma.useIt


#Segunda parte del main
puts "Esta es la segunda parte del main: "
num_jugadores = 4
dado2 = Dice.new()
emp_hang = [0,0]
emp_weap = [0,0,0]
emp_shiel = [0,0]
jug_prim = [0,0,0,0]
disp_prim = [0,0]
mov_esp = [0,0]

for index in(0..99)
  emp_hang[dado2.initWithNHangars]+= 1
  emp_weap[dado2.initWithNWeapons-1]+= 1
  emp_shiel[dado2.initWithNShields]+= 1
  jug_prim[dado2.whoStarts(4)]+= 1
  if (dado2.firstShot == GameCharacter::ENEMYSTARSHIP)
    disp_prim[0]+= 1
  else
    disp_prim[1]+= 1
  end
  if (dado2.spaceStationMoves(0.5) == true)
    mov_esp[0]+= 1
  else
    mov_esp[1]+= 1
  end

end
puts "Probabilidad de 1 y 0 en initWithNHangars"
puts emp_hang[0]
puts emp_hang[1]
puts "Probabilidad de 3, 1, 2 en initWithNWeapons"
puts emp_weap[0]
puts emp_weap[1]
puts emp_weap[2]
puts "Probabilidad de 1 y 0 en initWithNShields"
puts emp_shiel[0]
puts emp_shiel[1]
puts "Probabilidad de que empiece cada jugador"
puts jug_prim[0]
puts jug_prim[1]
puts jug_prim[2]
puts jug_prim[3]
puts "Probabilidad de que empiece primero enemy o ss"
puts disp_prim[0]
puts disp_prim[1]
puts "Probabilidad de que mueve la ss"
puts mov_esp[0]
puts mov_esp[1]

File: PRACTICA1/JAVA/deepspace/src/main/TestP1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package main;

import deepspace.Dice;
import deepspace.GameCharacter;
import deepspace.Loot;
import deepspace.ShieldBooster;
import deepspace.SuppliesPackage;
import deepspace.Weapon;
import deepspace.WeaponType;

/**
 *
 * @author celia
 */
public class TestP1 {
    public static void main(String [] args){
                // LOOT
		deepspace.Loot botin = new deepspace.Loot(8,4,3,5,6);
		System.out.println("Botín: ");
		System.out.println("Suministros: "+botin.getNSupplies());
		System.out.println("Armas : "+botin.getNWeapons());
		System.out.println("Escudos: "+botin.getNShields());
		System.out.println("Hangares: "+botin.getNHangars());
		System.out.println("Medallas: "+botin.getNMedals());

                // SUPPLIESPACKAGE
		deepspace.SuppliesPackage PaqueteSum = new SuppliesPackage(24.5f,22.5f,23.2f);
		System.out.println("Suministros: ");
		System.out.println("Municiones: "+PaqueteSum.getAmmoPower());
		System.out.println("Combustible: "+PaqueteSum.getFuelUnits());
		System.out.println("Poder de escudo: "+PaqueteSum.getShieldPower());

		SuppliesPackage PaqueteSumCopia = new SuppliesPackage(PaqueteSum);
		System.out.println("Suministros de copia: ");
		System.out.println("Municiones: "+PaqueteSumCopia.getAmmoPower());
		System.out.println("Combustible: "+PaqueteSumCopia.getFuelUnits());
		System.out.println("Poder de escudo: "+PaqueteSumCopia.getShieldPower());

                // SHIELDBOOSTER
		ShieldBooster escudo = new ShieldBooster("Escudo", 4.5f, 3);
		System.out.println("ShieldBooster: ");
                System.out.println("Boost: "+escudo.getBoost());
                System.out.println("Usos: "+escudo.getUses());
                System.out.println("Boost a partir de la función useIt: "+escudo.useIt());
                System.out.println("Usos después de useIt: "+escudo.getUses());
                
                ShieldBooster escudoCopia = new ShieldBooster(escudo);
                System.out.println("ShieldBooster de copia: ");
                System.out.println("Boost: "+escudo.getBoost());
                System.out.println("Usos: "+escudo.getUses());
                
                // WEAPON
                Weapon arma = new Weapon("armita", WeaponType.LASER,4 );
                
                System.out.println("Arma: ");
                System.out.println("Tipo: "+arma.getType());
                System.out.println("Usos: "+arma.getUses());
                System.out.println("Poder: "+arma.useIt());
                System.out.println("Usos después de useIt: "+arma.useIt());
                
                Weapon armaCopia = new Weapon(arma);
                System.out.println("Arma de copia: ");
                System.out.println("Tipo: "+armaCopia.getType());
                System.out.println("Usos: "+armaCopia.getUses());
                System.out.println("Poder: "+armaCopia.power());
                
                // DICE 
                // No sé si hace falta hacerlo o con poner Dice dado ya vale 
                Dice dado = new Dice();
                int[] hangars = new int[2];
                // Para comprobar que son cero 
                System.out.println("hangars[0]"+hangars[0]);
                System.out.println("hangars[1]"+hangars[1]);

                for (int i=0; i<100; i++){
                    hangars[dado.initWithNHangars()]++;
                }
                System.out.println("Probabilidad hangar 0= "+hangars[0]);
                System.out.println("Probabilidad hangar 1= "+hangars[1]);
                int[] weapons = new int[3];
                for (int i=0; i<100; i++){
                    weapons[dado.initWithNWeapons()-1]++;
                }
                System.out.println("Probabilidad weapons 1= "+weapons[0]);
                System.out.println("Probabilidad weapons 2= "+weapons[1]);
                System.out.println("Probabilidad weapons 3= "+weapons[2]);

                int[] shields = new int[2];
                for (int i=0; i<100; i++){
                    shields[dado.initWithNShields()]++;
                }
                System.out.println("Probabilidad shields 0= "+shields[0]);
                System.out.println("Probabilidad shields 1= "+shields[1]);

                int[] character = new int [2];
                
                for (int i=0; i<100; i++){
                   if(dado.firstShot()==GameCharacter.SPACESTATION){
                    character[0]++;
                   }
                   else {
                    character[1]++;
                   }   
                }
                
                System.out.println("Probabilidad SPACESTATION= "+character[0]);
                System.out.println("Probabilidad ENEMYSTARSHIP= "+character[1]);
                
                //probamos toString 
                System.out.println(dado.toString());
    }           
}
